# This file is part of HiGFXback

# requires
REQUIRES="libxslt-build python3-build"

pkg-config --exists --print-errors $REQUIRES || exit 1

# build
make -C src/lxml

# install
install -d $DESTDIR/lib/python3/lxml
install -m 644 src/lxml/_elementpath.py $DESTDIR/lib/python3/lxml
install src/lxml/_elementpath.so $DESTDIR/lib/python3/lxml
install src/lxml/etree.so $DESTDIR/lib/python3/lxml

# build.pc
install -d $DESTDIR/share/pkgconfig
cat > $DESTDIR/share/pkgconfig/lxml-build.pc << EOF
Name: lxml
Version: 4.5.2
Description: Pythonic XML processing library
Requires: $REQUIRES

exec=\\
/lib/python3/_elementpath.py \\
/lib/python3/_elementpath.so \\
/lib/python3/etree.so
EOF

exit
--- lxml-4.5.2.orig/src/lxml/Makefile
+++ lxml-4.5.2/src/lxml/Makefile
@@ -0,0 +1,10 @@
+all: _elementpath.so etree.so
+
+_elementpath.so: _elementpath.c
+	$(CC) -fPIC -shared -Wl,--no-undefined $^ -o $@ `pkg-config --cflags --libs python3`
+
+etree.so: etree.c
+	$(CC) -fPIC -shared -Wl,--no-undefined $^ -o $@ `pkg-config --cflags --libs libexslt python3` -Iincludes
+
+clean:
+	$(RM) *.so
